### PR TITLE
Check if window is valid before matchdelete

### PR DESCRIPTION
**Pull request by YuCao16**

### CODE REVIEW

**Comment by dvoytik** [comment]
Hi @YuCao16 ,

Thank you for your pull request!
I'm assuming you are fixing some bug. Could you please tell how this bug can be reproduced?

**Comment by YuCao16** [comment]
> Hi @YuCao16 ,
> 
> Thank you for your pull request! I'm assuming you are fixing some bug. Could you please tell how this bug can be reproduced?

My bad, my previous pr fix nothing. Here's the situation: If I highlight a word that is currently under the cursor, and then proceed to open a sidebar plugin like 'Symbol Outline', the same word appearing in the sidebar does not get highlighted. Additionally, invoking the `vim.fn.matchdelete` function leads to an error. I hope this clarifies the issue I'm encountering.

**Comment by YuCao16** [comment]
There should be better solutions, but this is probably the simplest one.

File: lua/hi-my-words/init.lua [hunk 1]
```diff
@@ -127,7 +127,7 @@ local function matchdel_all_windows(m_id)
   -- TODO: run in schedule?
   for _, w_id in ipairs(api.nvim_list_wins()) do
     api.nvim_win_call(w_id, function()
-      vim.fn.matchdelete(m_id)
+    pcall(vim.fn.matchdelete, m_id)
```
Comment (dvoytik) [inline lua/hi-my-words/init.lua]: I think an indent (two backspace characters) is missing.

**Review by dvoytik — CHANGES_REQUESTED** [review]
If you fix that indention problem, I'd merge.
Thanks.

**Comment by YuCao16** [comment]
@dvoytik fixed!

**Review by dvoytik — APPROVED** [review]

**Comment by dvoytik** [comment]
Thanks!

**Pull request merged**